test/crowdsale/TimedCrowdsale.test.js

```diff
@@ -73,5 +73,62 @@ contract('TimedCrowdsale', function ([_, investor, wallet, purchaser]) {
         await shouldFail.reverting(this.crowdsale.buyTokens(investor, { value: value, from: purchaser }));
       });
     });
+
+    describe('extending closing time', function () {
```

Comment:
The tests themselves are fine, but I think they can be made more readable by using `context` blocks, where each `context` has a `beforeEach` that sets up that context (advances time, etc.). E.g.:

```
describe _extendTime
  it reverts when reducing duration
  context before crowdsale start
    it extends end time
  context after crowdsale start
   it extends end time
  context after crowdsale end
   it reverts
```

What do you think?